### PR TITLE
Update gsidiag converter conventional data and satellite winds to comply with JEDI data convention 

### PR DESCRIPTION
**Pull request by emilyhcliu**
Update gsidiag converter to comply with JEDI data convention and update test reference for satellite radiance data

Address ioda-converters [issue #860](https://github.com/JCSDA-internal/ioda-converters/issues/860)

Please list the other repositories (if any) that this PR will require changes in (example below)
- [x] ioda  [PR #651](https://github.com/JCSDA-internal/ioda/pull/651)
- [x] https://github.com/JCSDA-internal/ioda/pull/664
- [x] https://github.com/JCSDA-internal/ioda/pull/686

### CODE REVIEW

**Comment by emilyhcliu** [comment]
@rhoneyager 
The variables in GsiQCWeight group can not pass the IODA validator.
The error message is the following:
```
  Variable GsiQCWeight/windEastward
   ERROR: Variable 'GsiQCWeight/windEastward' needs units, but the 'units' attribute does not exist in the file.
  Variable GsiQCWeight/windNorthward
   ERROR: Variable 'GsiQCWeight/windNorthward' needs units, but the 'units' attribute does not exist in the file.
```

File: src/gsi-ncdiag/gsi_ncdiag.py [hunk 1]
```diff
@@ -110,28 +132,28 @@
...
     'ZoverR': ('radar_dir3', 'float'),
     'Vterminal': ('vterminal', 'float'),
-    'SWCM_spec_type': ('satwind_spectral_type', 'float'),
-    'SAZA_sat_zen_angle': ('sensor_zenith_angle', 'float'),
-    'SCCF_chan_wavelen': ('channel_wavelength', 'float'),
-    'QI_with_FC': ('satwind_quality_ind_with_fc', 'float'),
-    'QI_without_FC': ('satwind_quality_ind_no_fc', 'float'),
+    'SWCM_spec_type': ('satelliteDerivedWindComputationMethod', 'float'),
+    'SAZA_sat_zen_angle': ('sensorZenithAngle', 'float'),
+    'SCCF_chan_wavelen': ('satelliteChannelCentreFrequency', 'float'),
```
Comment (rhoneyager) [inline src/gsi-ncdiag/gsi_ncdiag.py:137]: See comment in the ioda PR. Also, 'Centre' -> 'Center' or 'Central'.

**Comment by rhoneyager** [comment]
> The updated output files for conventional and satellite winds are: sfc_tv_obs_2018041500.nc4 satwind_obs_2018041500.nc4
> 
> @rhoneyager The variables in GsiQCWeight group can not pass the IODA validator. The error message is the following:
> 
> ```
>   Variable GsiQCWeight/windEastward
>    ERROR: Variable 'GsiQCWeight/windEastward' needs units, but the 'units' attribute does not exist in the file.
>   Variable GsiQCWeight/windNorthward
>    ERROR: Variable 'GsiQCWeight/windNorthward' needs units, but the 'units' attribute does not exist in the file.
> ```
> 
> The variables in this group should be float type and do not have dimension. In the converter, no unit is attached to this group, but the validator looks for the unit.

Can you clarify on not having dimensions? Do you mean no units, or do you mean that this group has a set of constant, single-element variables?

**Comment by rhoneyager** [comment]
Okay, I'll have a separate PR updating the validator to allow suppression of units for all variables in a particular group.

Besides that, there are conflicts between this branch and [feature/sprint-ioda-converters](https://github.com/JCSDA-internal/ioda-converters/tree/feature/sprint-ioda-converters) that should be addressed before merging.

**Comment by CoryMartin-NOAA** [comment]
Is this ready for review?

**Comment by emilyhcliu** [comment]
@Cory, I will work on this today!

**Comment by BenjaminRuston** [comment]
@emilyhcliu add me too as a reviewer please and let me know when to give it a try

**Comment by emilyhcliu** [comment]
Updated the ObsSpace.yaml 
See [IODA PR #686](https://github.com/JCSDA-internal/ioda/pull/686) 

The following relevant tests passed:
Run test as the following:
ctest -VV -R test_iodaconv_gsidiag_conv
ctest -VV -R test_iodaconv_gsidiag_conv_uv

The following relevant output test files passed the ioda validator:
sfc_tv_obs_2018041500.nc4
sfc_tsen_obs_2018041500.nc4
satwind_obs_2018041500.nc4

**Comment by BenjaminRuston** [comment]
@emilyhcliu the ctests are passing for me as well:

`ctest -R test_iodaconv_gsidiag_conv`
`ctest -R test_iodaconv_gsidiag_conv_uv`

but the validator is not:

`ctest -R iodaconv_validate_testoutput_satwind_obs_2018041500.nc4 -VV`

gives the error you noted before:

`ERROR: Variable 'GsiQCWeight/windEastward' `

am I missing something to suppress this

**Comment by BenjaminRuston** [comment]
also ran the `pycodestyle` validator on `gsi_ncdiag.py` and that passed.

so just the `units` which are unitless for the `GsiQCWeight`

**Comment by emilyhcliu** [comment]
This is what I got after running the validator:
```
Orion-login-4[282] eliu$ ./bin/ioda-validate.x ../ioda/share/ioda/yaml/validation/ObsSpace.yaml ./iodaconv/test/testrun/sfc_tv_obs_2018041500.nc4 
Reading YAML from ../ioda/share/ioda/yaml/validation/ObsSpace.yaml
Processing data file: ./iodaconv/test/testrun/sfc_tv_obs_2018041500.nc4
 Verifying that all required groups exist
 Verifying group PreUseFlag
 Verifying group ObsType
 Verifying group GsiAdjustObsError
 Verifying group ObsError
 Verifying group GsiQCWeight
 Verifying group MetaData
 Verifying group GsiFinalObsError
 Verifying group GsiEffectiveQC
 Verifying group ObsValue
 Verifying group GsiHofXBc
 Verifying group GsiHofX
 Verifying group GsiUseFlag
 Verifying group PreQC
 Verifying group /
 Verifying dimension names
 Verifying variable information
  Variable PreUseFlag/virtualTemperature
  Variable ObsType/virtualTemperature
  Variable GsiAdjustObsError/virtualTemperature
  Variable ObsError/virtualTemperature
  Variable GsiQCWeight/virtualTemperature
  Variable MetaData/latitude
  Variable MetaData/longitude
  Variable MetaData/pressure
  Variable MetaData/dateTime
  Variable MetaData/stationElevation
  Variable MetaData/height
  Variable MetaData/stationIdentification
  Variable GsiFinalObsError/virtualTemperature
  Variable GsiEffectiveQC/virtualTemperature
  Variable ObsValue/virtualTemperature
  Variable GsiHofXBc/virtualTemperature
  Variable GsiHofX/virtualTemperature
  Variable GsiUseFlag/virtualTemperature
  Variable PreQC/virtualTemperature
Final results:
  # errors:      0
  # warnings:    0
```

**Comment by emilyhcliu** [comment]
Here is the validator result for satwind file:
```
Orion-login-4[283] eliu$ ./bin/ioda-validate.x ../ioda/share/ioda/yaml/validation/ObsSpace.yaml ./iodaconv/test/testrun/satwind_obs_2018041500.nc4 
Reading YAML from ../ioda/share/ioda/yaml/validation/ObsSpace.yaml
Processing data file: ./iodaconv/test/testrun/satwind_obs_2018041500.nc4
 Verifying that all required groups exist
 Verifying group PreUseFlag
 Verifying group ObsType
 Verifying group GsiAdjustObsError
 Verifying group ObsError
 Verifying group GsiInputObsError
 Verifying group GsiQCWeight
 Verifying group MetaData
 Verifying group GsiFinalObsError
 Verifying group GsiEffectiveQC
 Verifying group ObsValue
 Verifying group GsiHofXBc
 Verifying group GsiHofX
 Verifying group GsiUseFlag
 Verifying group PreQC
 Verifying group /
 Verifying dimension names
 Verifying variable information
  Variable PreUseFlag/windEastward
  Variable PreUseFlag/windNorthward
  Variable ObsType/windEastward
  Variable ObsType/windNorthward
  Variable GsiAdjustObsError/windEastward
  Variable GsiAdjustObsError/windNorthward
  Variable ObsError/windEastward
  Variable ObsError/windNorthward
  Variable GsiInputObsError/windEastward
  Variable GsiInputObsError/windNorthward
  Variable GsiQCWeight/windEastward
  Variable GsiQCWeight/windNorthward
  Variable MetaData/stationIdentification
  Variable MetaData/sensorZenithAngle
  Variable MetaData/sensorCentralFrequency
  Variable MetaData/longitude
  Variable MetaData/stationElevation
  Variable MetaData/latitude
  Variable MetaData/percentConfidenceWithForecast
  Variable MetaData/pressure
  Variable MetaData/dateTime
  Variable MetaData/percentConfidenceWithoutForecast
  Variable MetaData/height
  Variable MetaData/windComputationMethod
  Variable GsiFinalObsError/windEastward
  Variable GsiFinalObsError/windNorthward
  Variable GsiEffectiveQC/windEastward
  Variable GsiEffectiveQC/windNorthward
  Variable ObsValue/windEastward
  Variable ObsValue/windNorthward
  Variable GsiHofXBc/windEastward
  Variable GsiHofXBc/windNorthward
  Variable GsiHofX/windEastward
  Variable GsiHofX/windNorthward
  Variable GsiUseFlag/windEastward
  Variable GsiUseFlag/windNorthward
  Variable PreQC/windEastward
  Variable PreQC/windNorthward
Final results:
  # errors:      0
  # warnings:    0
```

**Comment by emilyhcliu** [comment]
@BenjaminRuston 
I have PR for IODA with this IODA-converters PR.  https://github.com/JCSDA-internal/ioda/pull/686
You will need the update for ObsSpace.yaml from the IODA PR 686.

**Review by BenjaminRuston — APPROVED** [review]

**Review by srherbener — APPROVED** [review]

**Comment by srherbener** [comment]
The companion PR in ioda is ready for merge, and the relevant tests for this PR are passing. I will merge both PRs (ioda and this one) now. Thanks!

**Pull request merged**